libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -174,7 +174,7 @@ public static synchronized PublicKey getRSAPublicKey(String name, int length) {
      * @return RSA public key string.
      */
     public static synchronized String getRSAPublicString(String name, int length) {
-        final PublicKey publicKey = getRSAPublicKey(name, length);
+        final PublicKey publicKey = getRSAPublicKey(name.replaceAll("[^A-Za-z0-9]", ""), length);
```

Comment:
@Songroid Why are we still making changes here? We should be making this change in `PushService` where we are calling this method.